packages/utils/src/saml/pre-prod-tester.ts

```diff
@@ -424,21 +452,22 @@ async function queryDR(
       certChain: getEnvVarOrFail("CQ_ORG_CERTIFICATE_INTERMEDIATE_PRODUCTION"),
     };
 
-    const xmlResponses = createAndSignBulkDRRequests({
+    const signedRequests = createAndSignBulkDRRequests({
       bulkBodyData: [drRequest],
       samlCertsAndKeys,
     });
 
-    const response = await sendSignedDRRequests({
-      signedRequests: xmlResponses,
-      samlCertsAndKeys,
-      patientId: drRequest.patientId,
-      cxId: drRequest.cxId,
-    });
-
-    return processDrResponse({
-      drResponse: response[0],
+    const resultPromises = signedRequests.map(async (signedRequest, index) => {
+      return sendProcessRetryDrRequest({
+        signedRequest,
+        samlCertsAndKeys,
+        patientId: uuidv4(),
+        cxId: uuidv4(),
+        index,
+      });
     });
+    const results = await Promise.all(resultPromises);
+    return results[0];
```

Comment:
I know this is only for testing purpose - but broken tests eat a lot of one's time, so just to confirm we'll always have only one entry here?

Reply:
so long as the function signature doesnt change